### PR TITLE
guardrails bug fixes and IT for creating guardrails

### PR DESCRIPTION
**Pull request by jngz-es**
### Description
[Describe what this change achieves]
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: common/src/main/java/org/opensearch/ml/common/model/MLGuard.java [hunk 1]
```diff
@@ -159,4 +159,9 @@ public Boolean validateStopWordsSingleIndex(String input, String indexName, List
         }
         return hitStopWords.get();
     }
+
+    public enum Type {
```
Comment (dhrubo-os) [inline common/src/main/java/org/opensearch/ml/common/model/MLGuard.java:163]: May be we can add a from method like [this](https://github.com/opensearch-project/ml-commons/blob/main/common/src/main/java/org/opensearch/ml/common/FunctionName.java#L34)

Let's rename it to `MLGuardType`?

File: common/src/main/java/org/opensearch/ml/common/model/Guardrails.java [hunk 2]
```diff
@@ -22,26 +22,22 @@
 @Getter
 public class Guardrails implements ToXContentObject {
     public static final String TYPE_FIELD = "type";
-    public static final String ENGLISH_DETECTION_ENABLED_FIELD = "english_detection_enabled";
```
Comment (dhrubo-os) [inline common/src/main/java/org/opensearch/ml/common/model/Guardrails.java:25]: Just curious to know why are we removing this?

**Review by mingshl — COMMENTED** [review]
why decided to no longer use english detections?

File: common/src/main/java/org/opensearch/ml/common/model/MLGuard.java (hunk 1 above)
Comment (jngz-es) [reply common/src/main/java/org/opensearch/ml/common/model/MLGuard.java:163]: It's an inner class which is used as MLGuard.Type. So I don't think we need to add MLGuard prefix in the name.

File: common/src/main/java/org/opensearch/ml/common/model/Guardrails.java (hunk 2 above)
Comment (jngz-es) [reply common/src/main/java/org/opensearch/ml/common/model/Guardrails.java:25]: Yeah, I got comments before to remove it as we don't implement this feature right now. So to not confuse users, remove it now.

**Comment by jngz-es** [comment]
> why decided to no longer use english detections?

Just not introduce this feature in the first version. We will add it back in next release.

File: common/src/main/java/org/opensearch/ml/common/model/MLGuard.java (hunk 1 above)
Comment (dhrubo-os) [reply common/src/main/java/org/opensearch/ml/common/model/MLGuard.java:163]: can this MLGuard.Type be used in the Guardrails type also? OR these two types are different?

**Review by Zhangxunmt — APPROVED** [review]

**Review by mingshl — APPROVED** [review]

File: plugin/src/test/java/org/opensearch/ml/rest/RestMLGuardrailsIT.java [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+public class RestMLGuardrailsIT extends MLCommonsRestTestCase {
+
+    final String OPENAI_KEY = System.getenv("OPENAI_KEY");
+    final String COHERE_KEY = System.getenv("COHERE_KEY");
+
+    final String completionModelConnectorEntity = "{\n"
```
Comment (Zhangxunmt) [inline plugin/src/test/java/org/opensearch/ml/rest/RestMLGuardrailsIT.java:33]: Move this to Utility class of TestHelper under utils folder?

Comment (jngz-es) [reply plugin/src/test/java/org/opensearch/ml/rest/RestMLGuardrailsIT.java:33]: Yeah, will remove in next pr.

**Pull request merged**